docs/src/kmers.md

```diff
@@ -0,0 +1,187 @@
+```@meta
+CurrentModule = Kmers
+DocTestSetup = quote
+    using BioSequences
+    using Test
+    using Kmers
+end
+```
+
+## The `Kmer` type
+The central type of Kmers.jl is the `Kmer`.
+A `Kmer` is an immutable, bitstype `BioSequence`, with a length known at compile
+time. Compared to `LongSequence` in BioSequences.jl,
+this gives to one advantage, and comes with two disadvantages:
+* Kmers are much faster than `LongSequence`, as they can be stored in registers.
+* As kmers gets longer, the code gets increasingly inefficient, as the unrolling
+  and inlining of the immutable operations breaks down.
+* Since their length is part of their type, any operation that results in a kmer
+  whose length cannot be determined at compile time will be type unstable.
+  This includes slicing a kmer, pushing and popping it, and other operations.
+
+The `Kmer` type is (roughly) defined as
+```julia
+struct Kmer{A <: Alphabet, K, N} <: BioSequence{A}
+    x::NTuple{N, UInt}
+end
+```
+Where:
+* `A` is the `Alphabet` as defined in BioSequences.jl.
+* `K` is the length.
+* `N` is an extra type parameter derived from the first two,
+  which exists only because Julia does not allow computed type parameters.
+
+### Construction
+Kmers can be constructed from a `BioSequence` or `AbstractString` by explicitly
+specifying the length of the sequence:
+
+```jldoctest
+julia> Kmer{DNAAlphabet{2}, 5, 1}("TAGCT")
+DNA 5-mer:
+TAGCT
+```
+
+The final type parameter can be elided, in which case it will be inferred:
+
+```jldoctest
+julia> Kmer{DNAAlphabet{2}, 5}("TAGCT")
+DNA 5-mer:
+TAGCT
+```
+
+Kmers with alphabets `DNAAlphabet{2}`, `RNAAlphabet{2}` and `AminoAcidAlphabet`
+can be created with the type aliases `DNAKmer`, `RNAKmer` and `AAKmer`:
+
+```jldoctest
+julia> DNAKmer{3}("tag")
```

Comment:
Precisely - otherwise it'll be type unstable. String literals are expanded at parse time when constructing the AST, so instability is not an issue there.